packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/OutputLibrary.cs

```diff
@@ -49,15 +49,25 @@ private static TypeProvider[] BuildModels()
 
         protected virtual TypeProvider[] BuildTypeProviders()
         {
-            return
-            [
-                ..BuildEnums(),
-                ..BuildModels(),
+            var modelFactory = new ModelFactoryProvider(CodeModelPlugin.Instance.InputLibrary.InputNamespace.Models);
+
+            IEnumerable<TypeProvider> types = [
+                .. BuildEnums(),
+                .. BuildModels(),
                 new ChangeTrackingListDefinition(),
                 new ChangeTrackingDictionaryDefinition(),
                 new ArgumentDefinition(),
                 new OptionalDefinition(),
             ];
+
+            if (modelFactory.Methods.Count > 0)
```

Comment:
nit: can this logic be extracted into a BuildModelFactory method and just return empty array if count == 0?